imod/tests/test_formats/test_idf.py

```diff
@@ -112,19 +138,32 @@ def test_open_subdomains(test_da_subdomains, tmp_path):
     assert isinstance(da, xr.DataArray)
 
 
-def test_open_subdomains_species(test_da_subdomains_species, tmp_path):
-    subdomains = test_da_subdomains_species
+@parametrize_with_cases(
+    "subdomains,expected", cases=SubdomainCases, has_tag="no_species"
+)
+def test_open_subdomains_pattern_None(subdomains, expected, tmp_path):
+    """Read without provided pattern, function should interpet dimensions correctly"""
+    _save_subdomains_no_species(subdomains, tmp_path)
+    # Test with pattern is None
+    da = idf.open_subdomains(tmp_path / "subdomains_*.idf").load()
 
-    for i, subdomain in enumerate(subdomains):
-        for species, das in subdomain.groupby("species"):
-            for layer, da in das.groupby("layer"):
-                idf.write(
-                    tmp_path / f"subdomains_c{species}_20000101_l{layer}_p00{i}.idf", da
-                )
+    assert da.dims == ("time", "layer", "y", "x")
 
-    da = idf.open_subdomains(tmp_path / "subdomains_*.idf").load()
+    assert np.all(da.isel(time=0) == expected)
+
+
+@parametrize_with_cases("subdomains,expected", cases=SubdomainCases, has_tag="species")
+def test_open_subdomains_species(subdomains, expected, tmp_path):
+    _save_subdomains_species(subdomains, tmp_path)
+
+    # Test with pattern
+    pattern = r"{name}_c{species}_{time}_l{layer}_p{subdomain}"
 
-    assert np.all(da == 1.0)
+    da = idf.open_subdomains(tmp_path / "subdomains_*.idf", pattern=pattern).load()
+
+    assert da.dims == ("species", "time", "layer", "y", "x")
+
+    assert np.all(da.isel(time=0) == expected)
```

Comment:
you could also check the dtype of the result. IDF's can be single precision or double precision, do we want to have the dataset using the same type? or always double?

Reply:
Fixed it for non-equidistant grids!